### PR TITLE
Add remote certifcate monitoring infrastructure to keymasterd.

### PR DESCRIPTION
**Pull request by rgooch**

### CODE REVIEW

File: lib/constants/constants.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package constants
+
+const (
```
Comment (cviecco) [inline lib/constants/constants.go]: None of these are consts. There are default values, but the serving values are actually defined by the configuration.

File: lib/constants/constants.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+	KeymasterAdminPortNumber   = 6920
+	KeymasterServicePortNumber = 443
+
+	CertmonPortNumber = 6921
```
Comment (cviecco) [inline lib/constants/constants.go]: certmonPortNumber? ... I see that you (just like we had discussed) are using the controlPort in a separate path.
I would drop this file

File: keymasterd/certnotifier/impl.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+func (n *CertNotifier) serveHTTP(w http.ResponseWriter, req *http.Request) {
+	if req.Method != "CONNECT" {
+		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+	if hijacker, ok := w.(http.Hijacker); !ok {
+		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+		w.WriteHeader(http.StatusMethodNotAllowed)
```
Comment (cviecco) [inline keymasterd/certnotifier/impl.go]: I think that is an InternalError, not a statusmethodNotAlowed

File: keymasterd/certnotifier/impl.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		return
+	} else {
```
Comment (cviecco) [inline keymasterd/certnotifier/impl.go]: I think is more clear to declare the hijacker and then avoid this else block

File: keymasterd/certnotifier/impl.go [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+		conn, bufRw, err := hijacker.Hijack()
+		if err != nil {
+			n.logger.Println("certmon hijacking ", req.RemoteAddr, ": ",
```
Comment (cviecco) [inline keymasterd/certnotifier/impl.go]: Also need to send a http.StatusInternalServerError to the client

File: lib/constants/constants.go (hunk 1 above)
Comment (rgooch) [reply lib/constants/constants.go]: How about I prepend these with `Default`?

File: lib/constants/constants.go (hunk 2 above)
Comment (rgooch) [reply lib/constants/constants.go]: `CertmonPortNumber` will be used by `keymaster-certmond`, which is coming in a followup PR. I'll send that once you've added the `Publish*()` method calls, so I can test.

File: keymasterd/certnotifier/impl.go (hunk 3 above)
Comment (rgooch) [reply keymasterd/certnotifier/impl.go]: Will fix.

File: keymasterd/certnotifier/impl.go (hunk 4 above)
Comment (rgooch) [reply keymasterd/certnotifier/impl.go]: Will fix.

File: keymasterd/certnotifier/impl.go (hunk 5 above)
Comment (rgooch) [reply keymasterd/certnotifier/impl.go]: Will fix.

**Comment by rgooch** [comment]
I've pushed fixed. PTAL.

**Comment by cviecco** [comment]
I would rather not having those folded into the source tree until they
are actually refered by something.

Camilo

**Comment by rgooch** [comment]
Sigh. lib/constants removed. PTAL.

**Pull request merged**